cpp/src/sampling/random_walks_impl.cuh

```diff
@@ -483,7 +555,7 @@ biased_random_walks(raft::handle_t const& handle,
     std::optional<edge_property_view_t<edge_t, weight_t const*>>{edge_weight_view},
     start_vertices,
     max_length,
-    detail::biased_selector<weight_t>{(seed == 0 ? detail::get_current_time_nanoseconds() : seed)});
+    detail::biased_selector<weight_t>((seed == 0 ? detail::get_current_time_nanoseconds() : seed)));
```

Comment:
Our API generally takes `rng_state` instead of seed (we have few functions taking seed instead, and we need to fix them as well).

See https://github.com/rapidsai/cugraph/blob/branch-24.08/cpp/include/cugraph/algorithms.hpp and search for rng_state. Random walk functions need to be updated to follow this convention.

e.g. Leiden takes `raft::random::RngState& rng_state` instead of seed.

And see https://github.com/rapidsai/cugraph/blob/branch-24.08/cpp/tests/prims/mg_per_v_random_select_transform_outgoing_e.cu#L173

This ensures that every GPU will use different seeds (so use different sequences of random numbers).